naptime/src/main/scala/org/coursera/naptime/actions/RestAction.scala

```diff
@@ -85,12 +87,16 @@ trait RestAction[RACType, AuthType, BodyType, KeyType, ResourceType, ResponseTyp
 
   private[naptime] def safeApply(
       context: RestContext[AuthType, BodyType]): Future[RestResponse[ResponseType]] = {
-    try {
-      apply(context) recover errorHandler
-    } catch {
-      case NonFatal(e) if errorHandler.isDefinedAt(e) =>
-        errorHandler.andThen(Future.successful)(e)
-    }
+    Try(apply(context))
+      .recover {
+        case e: Throwable => Future.failed(e)
+      }
+      .get
```

Comment:
This seems strange (`.recover.get.recover.recover`), can we refactor this a bit?

I'm a bit confused as to why we need both `Try` and `Future` when both offers error handling.